docker/daemon.go

```diff
@@ -47,6 +47,9 @@ func NewDaemonCli() *DaemonCli {
 	daemonConfig := new(daemon.Config)
 	daemonConfig.LogConfig.Config = make(map[string]string)
 	daemonConfig.ClusterOpts = make(map[string]string)
+
+	daemonFlags.StringVar(&daemonConfig.ConfigurationFile, []string{"-config"}, "", "Daemon configuration file")
```

Comment:
What do you think about having a default path, and then doing the fsnotify watch even if it doesn't exist so we can catch creation of the file after the daemon starts up?